paddle.js

```diff
@@ -0,0 +1,50 @@
+class Paddle {
+  constructor(canvas, ctx, width, height, x, y, speed, color) {
+    this.canvas = canvas;
+    this.ctx = ctx;
+    this.width = width;
+    this.height = height;
+    this.x = x;
+    this.y = y;
+    this.speed = speed;
+    this.color = color;
+    this.direction = 0;
+  }
+
+  getTopEdge() {
+    return this.y;
+  }
+
+  getBottomEdge() {
+    return this.x + this.height;
+  }
+
+  getLeftEdge() {
+    return this.x;
+  }
+
+  getRightEdge() {
+    return this.x + this.width;
+  }
+
+  move() {
+    const left = this.getLeftEdge();
+    const right = this.getRightEdge();
+
+    if (
```

Comment:
`if (left >= 0 && right <= this.canvas.width)`

you don't need direction in this condition at all

Reply:
I need this condition, otherwise the paddle will get stuck once I reach the ends of left and right.

Reply:
it gets stuck because you're out of bounds. you should actually be using the future values to determine whether or not the paddle should move. i.e nextX = left + (direction * speed)

Reply:
why is it more optimal to use future values instead of setting the constraint as is?

Reply:
my response was to your issue 

> I need this condition, otherwise the paddle will get stuck once I reach the ends of left and right.

the reason the paddle gets stuck is because you are checking the criteria before the paddle moves, then the paddle moves out of bounds and can no longer move left or right. instead, you need to use the future coordinates in the constraint condition:

if futureX >= 0 && futureRightEdge <= canvas width

Reply:
you initially commented that i do not need direction, which is why i brought up what would happen if i were to not use direction in my constraint. the future values you suggested is calculated using direction.

Reply:
this isn't resolved